tests/Unit/RuleSet/DeclarationBlockTest.php

```diff
@@ -48,4 +52,78 @@ public function implementsPositionable(): void
     {
         self::assertInstanceOf(Positionable::class, $this->subject);
     }
+
+    /**
+     * @return array<non-empty-string, array{0: non-empty-string}>
+     */
+    public static function provideSelector(): array
+    {
+        return [
+            'type' => ['body'],
+            'class' => ['.teapot'],
+            'type & class' => ['img.teapot'],
+            'id' => ['#my-mug'],
+            'type & id' => ['h2#my-mug'],
+            'pseudo-class' => [':hover'],
+            'type & pseudo-class' => ['a:hover'],
+            '`not`' => [':not(#your-mug)'],
+            'pseudo-element' => ['::before'],
+            'attribute with `"`' => ['[alt="{}()[]\\"\'"]'],
+            'attribute with `\'`' => ['[alt=\'{}()[]"\\\'\']'],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param non-empty-string $selector
+     *
+     * @dataProvider provideSelector
+     */
+    public function parsesSingleSelector(string $selector): void
+    {
+        $subject = DeclarationBlock::parse(new ParserState($selector . '{}', Settings::create()));
+
+        self::assertNotNull($subject);
+        self::assertSame([$selector], self::getSelectorsAsStrings($subject));
+    }
+
+    /**
+     * @return DataProvider<non-empty-string, array{0: non-empty-string, 1: non-empty-string}>
+     */
+    public static function provideTwoSelectors(): DataProvider
+    {
+        return DataProvider::cross(self::provideSelector(), self::provideSelector());
+    }
+
+    /**
+     * @test
+     *
+     * @param non-empty-string $firstSelector
+     * @param non-empty-string $secondSelector
+     *
+     * @dataProvider provideTwoSelectors
+     */
+    public function parsesTwoCommaSeparatedSelectors(string $firstSelector, string $secondSelector): void
+    {
+        $joinedSelectors = $firstSelector . ',' . $secondSelector;
```

Comment:
Usually, a space is used after the comma AFAIK. Should we add a space here as well?

Reply:
It's arbitrary.  A space is added in unminified CSS for readability.  I've added it.